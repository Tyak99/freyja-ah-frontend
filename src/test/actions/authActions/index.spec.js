import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import nock from 'nock';
import {
  ResetPassword, loading, passwordResetSuccess, passwordResetError,
} from '../../../../store/actions/authActions/resetPassword';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);


describe('async actions', () => {
  let store;
  beforeEach(() => {
    store = mockStore({});
  });
  afterEach(() => {
    nock.cleanAll();
  });

  it('creates PASSWORD_RESET_SUCCESS when fetching to-dos has been done', () => {
    nock('https://freyja-ah-backend.herokuapp.com')
      .post('/api/users/reset')
      .reply(200, { message: 'Please check your email' });

    return store.dispatch(ResetPassword('mattimobolaji@gmail.com'))
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });

  it('creates PASSWORD_RESET_ERROR when fetching to-dos has been done', () => {
    nock('https://freyja-ah-backend.herokuapp.com')
      .post('/api/users/reset')
      .reply(401, { error: 'invalid email' });

    return store.dispatch(ResetPassword('mattimobolaji@gmail.commm'))
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });


  it('creates PASSWORD_RESET_SUCCESS when reset password action was successful', () => {
    store.dispatch(passwordResetSuccess('Please check your email'));
    expect(store.getActions()).toMatchSnapshot();
  });

  it('creates PASSWORD_RESET_ERROR when reset password action was not successful', () => {
    store.dispatch(passwordResetError('invalid Email'));
    expect(store.getActions()).toMatchSnapshot();
  });

  it('updates loading to true in state', () => {
    store.dispatch(loading());
    expect(store.getActions()).toMatchSnapshot();
  });
});
