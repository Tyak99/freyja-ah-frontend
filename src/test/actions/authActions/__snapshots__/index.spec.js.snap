// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async actions creates PASSWORD_RESET_ERROR when fetching to-dos has been done 1`] = `
Array [
  Object {
    "type": "LOADING",
  },
  Object {
    "payload": undefined,
    "type": "PASSWORD_RESET_ERROR",
  },
]
`;

exports[`async actions creates PASSWORD_RESET_ERROR when reset password action was not successful 1`] = `
Array [
  Object {
    "payload": "invalid Email",
    "type": "PASSWORD_RESET_ERROR",
  },
]
`;

exports[`async actions creates PASSWORD_RESET_SUCCESS when fetching to-dos has been done 1`] = `
Array [
  Object {
    "type": "LOADING",
  },
  Object {
    "payload": "Please check your email",
    "type": "PASSWORD_RESET_SUCCESS",
  },
]
`;

exports[`async actions creates PASSWORD_RESET_SUCCESS when reset password action was successful 1`] = `
Array [
  Object {
    "payload": "Please check your email",
    "type": "PASSWORD_RESET_SUCCESS",
  },
]
`;

exports[`async actions updates loading to true in state 1`] = `
Array [
  Object {
    "type": "LOADING",
  },
]
`;
